src/vitamin.cpp

```diff
@@ -47,6 +47,56 @@ int vitamin::severity( int qty ) const
     return 0;
 }
 
+std::string vitamin::get_string_level( int qty ) const
+{
+    float norm_level = 1.0f;
+    if( type_ == vitamin_type::VITAMIN ) {
+        if( id_ == "vitA" ) {
+            norm_level = 2.0f;
+        } else if( id_ == "vitB" ) {
+            norm_level = 2.75f;
+        } else if( id_ == "vitC" ) {
+            norm_level = 2.2f;
+        } else if( id_ == "iron" ) {
+            norm_level = 1.23f;
+        } else if( id_ == "calcium" ) {
+            norm_level = 1.1f;
+        }
+        const float v_level = norm_level * qty / disease_[0].first;
```

Comment:
Some vitamins don't have anything in `disease_`, this will be attempting to access a non-existing member of a vector.
Additionally, this means that vitamins that use `disease_excess_`  will always get the default message.
Also, this element is supposed to be loaded as 0 if it is not specified. I'm not sure if that would happen in practice, but there could be potential for a divide by zero here.

Reply:
Fair point, thanks.